relay-server/src/actors/store.rs

```diff
@@ -1049,3 +1103,162 @@ impl Message for KafkaMessage {
 fn is_slow_item(item: &Item) -> bool {
     item.ty() == &ItemType::Attachment || item.ty() == &ItemType::UserReport
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use relay_common::ProjectKey;
+
+    /// Helper function to get the arguments for the `fn extract_kafka_messages(...)` method.
+    fn test_arguments_extract_kafka_msgs() -> (Instant, EventId, Scoping, Vec<ChunkedAttachment>) {
+        let start_time = Instant::now();
+        let event_id = EventId::new();
+        let scoping = Scoping {
+            organization_id: 42,
+            project_id: ProjectId::new(21),
+            project_key: ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap(),
+            key_id: Some(17),
+        };
+
+        let attachment_vec = {
+            let item = Item::new(ItemType::Attachment);
+
+            vec![ChunkedAttachment {
+                id: Uuid::new_v4().to_string(),
+                name: UNNAMED_ATTACHMENT.to_owned(),
+                content_type: item
+                    .content_type()
+                    .map(|content_type| content_type.as_str().to_owned()),
+                attachment_type: item.attachment_type().cloned().unwrap_or_default(),
+                chunks: 0,
+                size: None,
+                rate_limited: Some(item.rate_limited()),
+            }]
+        };
+
+        (start_time, event_id, scoping, attachment_vec)
+    }
+
+    #[test]
+    fn test_return_attachments_when_missing_event_item() {
+        let (start_time, event_id, scoping, attachment_vec) = test_arguments_extract_kafka_msgs();
+
+        let mut kafka_messages = StoreService::extract_kafka_messages(
+            None,
+            event_id,
+            scoping,
+            start_time,
+            None,
+            attachment_vec,
+        );
+
+        assert!(kafka_messages.any(|msg| {
+            if let KafkaMessage::Attachment(_) = msg {
+                return true;
+            }
+            false
+        }));
```

Comment:
They're all of the same type, right?

```suggestion
        assert!(kafka_messages.all(|msg| matches!(msg, KafkaMessage::Attachment(_)));
```

Reply:
yeah good point, that way seems more explicit